src/routes/clients.js

```diff
@@ -61,14 +71,26 @@ router.post('/', (req, res) => {
       case 'name':
         errno = ERRNO_INVALID_API_CLIENT_NAME;
         break;
+      case 'permissions':
+        errno = ERRNO_INVALID_API_CLIENT_PERMISSION;
+        break;
       default:
         errno = ERRNO_BAD_REQUEST;
     }
     return ApiError(res, 400, errno, BAD_REQUEST);
   }
 
   db().then(models => {
-    models.Clients.create(req.body).then(client => {
+    let client;
+    models.sequelize.transaction(transaction => {
```

Comment:
OK, I was confused with indentation because I thought the response was done on the upper chain. That said, I think it would be better at this location ;)
I mean:
```javascript
db().then(models => {
  return transaction(transaction => {
    return create().then(() => addPermissions())
  });
}).then(() => res.status(XXX).send())
.catch(...)

```